services/src/main/java/org/keycloak/credential/PasswordCredentialProvider.java

```diff
@@ -164,7 +164,7 @@ public void disableCredentialType(RealmModel realm, UserModel user, String crede
 
     @Override
     public boolean supportsCredentialType(String credentialType) {
-        return credentialType.equals(CredentialModel.PASSWORD);
+        return credentialType == null || credentialType.equals(CredentialModel.PASSWORD);
```

Comment:
FYI ` CredentialModel.PASSWORD.equals(credentialType)` in that order can't throw a null pointer exception, usually always the constants first because equals accept "null" but null don't have equals function.

Reply:
I think _true_ should be returned if _credentialType_ is null, that is _credentialType_ is not explicitly specified. In other words, default _credentialType_ is _CredentialModel.PASSWORD_.

Reply:
It what I mean, that was just fyi, `CredentialModel.PASSWORD.equals(credentialType)` is the very same thing than `credentialType == null || credentialType.equals(CredentialModel.PASSWORD);` No changes required tho, it's just shorter.

Reply:
Ok, I see what you mean, I though you was trying to avoid a null pointer exception.